File: Patterns/code/java/DesignPatterns/src/main/java/decorator/MushroomDecorator.java
package decorator;

class MushroomDecorator extends PizzaDecorator {
    public MushroomDecorator(Pizza pizza) {
        super(pizza);
    }

    @Override
    public void bake() {
        super.bake();
        System.out.println("Adding mushrooms");
    }
}

File: code/typescript/decorator/MushroomDecorator.ts
import { PizzaDecorator } from "./PizzaDecorator";
import { Pizza } from "./Pizza";

export class MushroomDecorator extends PizzaDecorator {
    constructor(pizza: Pizza) {
        super(pizza);
    }

    bake(): void {
        super.bake();
        console.log("Adding mushrooms");
    }
}
